src/ucs/memory/memory_type.h

```diff
@@ -40,6 +40,7 @@ typedef enum ucs_memory_type {
     UCS_MEMORY_TYPE_CUDA_MANAGED,  /**< NVIDIA CUDA managed (or unified) memory */
     UCS_MEMORY_TYPE_ROCM,          /**< AMD ROCM memory */
     UCS_MEMORY_TYPE_ROCM_MANAGED,  /**< AMD ROCM managed system memory */
+    UCS_MEMORY_TYPE_RDMA,          /**< Infiniband device memory */
```

Comment:
Yikes, this breaks backward compatibility, since it changes the value of "UCS_MEMORY_TYPE_UNKNOWN"
Anyway, let's see if defining new memory type works, we may add if after "UNKNOWN"

Reply:
@yosefe  If we eventually decide to break it, maybe better move `UCS_MEMORY_TYPE_UNKNOWN` to index 0 (and decouple it from `_LAST`? So that it never breaks in the future.